README.md

```diff
@@ -92,43 +92,133 @@ Requirements:
 
+The nginx-datadog module **depends on two git submodules**:
+  - [libddwaf](https://github.com/DataDog/libddwaf)
+  - [dd-trace-cpp](https://github.com/DataDog/dd-trace-cpp)
+
+Building the module with any of the below commands will init and update them. Otherwise you can do it yourself with:
+```bash
+git submodule update --init --recursive
+```
+
+The easiest way to build the project is through [Docker][2].
+
+Build module using Docker
+---------------
+### Linux, MacOS AMD64
 ```shell
-NGINX_VERSION=1.25.2 make build
+NGINX_VERSION=1.25.2 make build-musl
 ```
 
+### MacOS with Apple Silicon
+```shell
+NGINX_VERSION=1.25.2 ARCH=aarch64 make build-musl
+```
+The resulting nginx module is `.musl-build\ngx_http_datadog_module.so`
+
+The `build-musl` target builds against musl and libc++ a glibc-compatible
+module. The Dockerfile for the docker image used in the process can be found in
+[build_env/Dockerfile](./build_env/Dockerfile).
+
+### AppSec-supporting module
+
+
 You can set the environment variable `WAF` to `ON` to build an AppSec-supporting
 module:
 
 ```shell
-WAF=ON NGINX_VERSION=1.25.2 make build
+WAF=ON NGINX_VERSION=1.25.2 make build-musl
+```
+
+## Build OpenResty compatible module using Docker
+
+### Linux, MacOS AMD64
+```shell
+RESTY_VERSION=1.27.1.1 make build-openresty
+```
+
+### MacOS with Apple Silicon
+```shell
+RESTY_VERSION=1.27.1.1 ARCH=aarch64 make build-openresty
+```
+
+The resulting nginx module is `.openresty-build/ngx\_http\_datadog\_module.so`
+
+Build in Local Environment
+--------------------------
+```shell
+NGINX_VERSION=1.25.2 make build
 ```
 
-The resulting nginx module is `.build/ngx\_http\_datadog\_module.so`
+The resulting nginx module is `.build/ngx\_http\_datadog\_module.so`.
+
+If you encounter some **difficulties** building the module on **MacOS**, please look at the **troubleshooting section**.
 
 The `build` target does the following:
 
 - Download a source release of nginx based on the `NGINX\_VERSION` environment variable.
 - Initialize the source tree of `dd-trace-cpp` as a git submodule.
+- Initialize the source tree of `libddwaf`as a git submodule.
 - Build `dd-trace-cpp` and the Datadog nginx module together using
   CMake.
 
 `make clean` deletes CMake's build directory. `make clobber` deletes
 everything done by the build.
 
-Build in Docker
----------------
+Testing
+-------
+
+The makefile contains two target for testing:
+- build-and-test: builds and use the resultant module for testing
+- test: use the existing built module for testing
+
+To run one or the other, you can use:
+
+### Linux, MacOS AMD64
 ```shell
-make build-musl
+NGINX_VERSION=1.25.2 make build-and-test
+```
+### MacOS with Apple Silicon
+```shell
+NGINX_VERSION=1.25.2 ARCH=aarch64 make build-and-test
+```
+By default, it will launch the test on the `nginx:${NGINX_VERSION}-alpine` docker image.
+If you want to use another nginx image you can use:
+```shell
+BASE_IMAGE=nginx:1.25.2-jammy make build-and-test
+```
+#
```

Comment:
leftover?

Reply:
It is to show an exemple on how you can specify a base_image. I replaced it with an existing image